provisioning/service/samples/create_tpm_enrollment_with_token_credential.js

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+const { ClientSecretCredential } = require("@azure/identity");
+var provisioningServiceClient = require('azure-iot-provisioning-service').ProvisioningServiceClient;
+var uuid = require('uuid');
+
+var argv = require('yargs')
+  .usage('Usage: $0 --hostname <DPS HOST NAME> --endorsementKey <ENDORSMENT KEY> --tenantid <AAD TENANT ID> --clientid <AAD CLIENT ID> --clientsecret <AAD_CLIENT_SECRET>')
+  .option('hostname', {
+    alias: 'h',
+    describe: 'The host name for the Device Provisioning instance',
+    type: 'string',
+    demandOption: true
+  })
+  .option('tenantid', {
+    alias: 't',
+    describe: 'The Azure Active Directory tenant (directory) Id',
+    type: 'string',
+    demandOption: true
+  })
+.option('clientid', {
+    alias: 'c',
+    describe: 'The client Id of the Azure Active Directory application',
+    type: 'string',
+    demandOption: true
+  })
+  .option('clientsecret', {
+    alias: 's',
+    describe: 'A client secret that was generated for the application Registration used to authenticate the client',
+    type: 'string',
+    demandOption: true
+  })
+  .argv;
+
+var credential = new ClientSecretCredential(argv.tenantid, argv.clientid, argv.clientsecret);
+var serviceClient = provisioningServiceClient.fromTokenCredential(argv.hostname, credential);
+
+var enrollment = {
+  registrationId: 'sample_enrollment-' + uuid.v1(),
+  attestation: {
+    type: 'symmetricKey',
+    symmetricKey: {
+      primaryKey: '',
+      secondaryKey: ''
+    }
+  },
+  capabilities: {
+    iotEdge: false
+  }
+};
```

Comment:
A little confused here. The sample is supposed to for RBAC, but the enrollment options here says symmetricKey.

Reply:
As far as I understood the feature, it was only to allow RBAC for the authentication for the actual requests themselves (as opposed to a connection string or Sas credential), and the device auth (symmetric key, x509, etc) are separate

Reply:
Sorry, yeah you're right. For the primary key and secondary key fields, I think it'd be more clear to the customer if instead of having an empty string do `<primary key>` and `<secondary key>`